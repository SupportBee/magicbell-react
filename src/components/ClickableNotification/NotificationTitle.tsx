/** @jsx jsx */
import { css, jsx } from '@emotion/react';
import INotification from '@magicbell/react-headless/dist/types/INotification';

export interface Props {
  notification: INotification;
}

/**
 * Component that renders the title of a notification.
 *
 * @example
 * <NotificationTitle notification={notification} />
 */
export default function NotificationTitle({ notification }: Props) {
  const { title } = notification;

  return (
    <p
      title={title}
      css={css`
        cursor: inherit;
        font-weight: 500 !important;
        line-height: 1.2 !important;
        word-break: break-word !important;
      `}
    >
      {title}
    </p>
  );
}
