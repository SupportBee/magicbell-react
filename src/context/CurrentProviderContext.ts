import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
