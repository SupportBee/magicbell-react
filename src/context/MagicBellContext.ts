import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
