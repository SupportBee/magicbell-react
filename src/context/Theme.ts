export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
