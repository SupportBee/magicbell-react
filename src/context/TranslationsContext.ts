import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
