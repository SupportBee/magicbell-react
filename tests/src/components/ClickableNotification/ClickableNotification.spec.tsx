import { INotification, useNotificationStoresCollection } from '@magicbell/react-headless';
import { act, render, RenderResult } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import userEvent from '@testing-library/user-event';
import dayjs from 'dayjs';
import { Response, Server } from 'miragejs';
import React from 'react';
import Sinon from 'sinon';
import ClickableNotification from '../../../../src/components/ClickableNotification';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';
import { sampleNotification } from '../../../factories/NotificationFactory';

describe('components', () => {
  describe('ClickableNotification', () => {
    let clock: Sinon.SinonFakeTimers;
    let notification = sampleNotification as any;
    let onClick: (notification: INotification) => void;
    let view: RenderResult;

    beforeEach(async () => {
      clock = Sinon.useFakeTimers(1605412800000);
      onClick = jest.fn();

      await act(async () => {
        view = render(
          <MagicBellThemeProvider value={defaultTheme}>
            <ClickableNotification notification={notification} onClick={onClick} />
          </MagicBellThemeProvider>,
        );
      });
    });

    afterEach(() => {
      clock.restore();
    });

    describe('render', () => {
      it('renders the title and content of the notification', () => {
        expect(view.container).toMatchSnapshot();
      });

      describe('the notification has no content', () => {
        it('renders the title of the notification', () => {
          view.rerender(
            <MagicBellThemeProvider value={defaultTheme}>
              <ClickableNotification
                notification={{ ...notification, content: null }}
                onClick={onClick}
              />
            </MagicBellThemeProvider>,
          );

          expect(view.container).toMatchSnapshot();
        });
      });

      describe('theming', () => {
        describe('the notification is unseen', () => {
          it('renders the notification with the unseen theme variation', () => {
            view.rerender(
              <MagicBellThemeProvider value={defaultTheme}>
                <ClickableNotification
                  notification={{ ...notification, seenAt: null, readAt: null }}
                  onClick={onClick}
                />
              </MagicBellThemeProvider>,
            );

            expect(view.container).toMatchSnapshot();
          });
        });

        describe('the notification is seen and read', () => {
          it('renders the notification with the default theme variation', async () => {
            view.rerender(
              <MagicBellThemeProvider value={defaultTheme}>
                <ClickableNotification
                  notification={{
                    ...notification,
                    seenAt: dayjs().unix(),
                    readAt: dayjs().unix(),
                  }}
                  onClick={onClick}
                />
              </MagicBellThemeProvider>,
            );

            expect(await view.findByText('This is a good content')).toBeInTheDocument();
            expect(view.container).toMatchSnapshot();
          });
        });
      });
    });

    describe('.handleClick', () => {
      let server: any;

      beforeEach(() => {
        server = new Server({
          environment: 'test',
          urlPrefix: 'https://api.magicbell.com',
          timing: 50,
        });

        server.post(`/notifications/${notification.id}/read`, new Response(201, {}, ''));
      });

      afterEach(() => {
        server.shutdown();
      });

      it('passes the notification object to the onClick callback', () => {
        const component = view.getByTestId('clickable-notification');
        userEvent.click(component);

        expect(onClick).toHaveBeenCalledTimes(1);
      });

      it.skip('marks the notification as read', () => {
        const { result } = renderHook(() => useNotificationStoresCollection());
        const spy = jest.spyOn(result.current, 'markNotificationAsRead');
        userEvent.click(view.getByTestId('clickable-notification'));

        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith(notification.id);
      });
    });

    describe('default handler', () => {
      beforeEach(async () => {
        await act(async () => {
          view.rerender(
            <MagicBellThemeProvider value={defaultTheme}>
              <ClickableNotification notification={notification} />
            </MagicBellThemeProvider>,
          );
        });
      });

      describe('the notification has an actionUrl', () => {
        it('opens the action url in the same window', () => {
          userEvent.click(view.getByText(notification.title));

          expect(global.open).toHaveBeenCalledTimes(1);
          expect(global.open).toHaveBeenCalledWith(notification.actionUrl, '_self');
        });
      });
    });

    describe('on unmount', () => {
      it.skip('marks the notification as seen', () => {
        const { result } = renderHook(() => useNotificationStoresCollection());
        const spy = jest.spyOn(result.current, 'markNotificationAsSeen');

        view.unmount();

        expect(spy).toHaveBeenCalledTimes(1);
        expect(spy).toHaveBeenCalledWith(notification.id);
      });
    });
  });
});
