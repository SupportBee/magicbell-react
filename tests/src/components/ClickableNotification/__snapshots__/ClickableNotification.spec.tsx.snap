// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ClickableNotification render renders the title and content of the notification 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  background-color: rgba(52, 152, 244, 0)!important;
  border-radius: 8px;
  color: #3A424D!important;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px;
  margin: 4px!important;
  overflow: hidden;
  text-align: left!important;
  text-transform: none;
  -webkit-transition: background-color 300ms ease-out;
  transition: background-color 300ms ease-out;
  padding: 16px 20px!important;
  min-height: 32px;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:hover {
  background-color: rgba(49, 144, 232, 0.05)!important;
}

.emotion-0>button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -16px;
  text-align: inherit!important;
}

.emotion-0>button>div {
  margin: 0 0 0 16px;
}

.emotion-0>button>div:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 {
  cursor: inherit;
  font-weight: 500!important;
  line-height: 1.2!important;
  word-break: break-word!important;
}

.emotion-2 {
  color: inherit!important;
  cursor: inherit;
  margin: 0;
  line-height: 1.2;
  font-size: 0.865em!important;
  margin-top: 1em!important;
  word-break: break-word!important;
}

.emotion-3 {
  padding: 0.5em 0 0.5em 0.25em!important;
}

.emotion-4 {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  -webkit-align-items: center!important;
  -webkit-box-align: center!important;
  -ms-flex-align: center!important;
  align-items: center!important;
  padding: 0.5em 0 0.5em 0.25em!important;
  color: #3498F4!important;
}

.emotion-4 svg {
  height: 1.5em;
  fill: currentColor;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      data-testid="clickable-notification"
    >
      <div>
        <p
          class="emotion-1"
          title="New Comment: Tables in knowledgebase"
        >
          New Comment: Tables in knowledgebase
        </p>
        <article
          class="emotion-2"
        >
          <p>
            This is a good content
          </p>
        </article>
      </div>
      <div
        class="emotion-3"
      >
        <div
          aria-expanded="false"
        >
          9mo
        </div>
      </div>
    </button>
    <div
      aria-expanded="false"
      class="emotion-4"
      data-testid="icon"
    >
      <svg
        viewBox="0 0 512 512"
      >
        <circle
          cx="256"
          cy="256"
          r="64"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`components ClickableNotification render the notification has no content renders the title of the notification 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  background-color: rgba(52, 152, 244, 0)!important;
  border-radius: 8px;
  color: #3A424D!important;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px;
  margin: 4px!important;
  overflow: hidden;
  text-align: left!important;
  text-transform: none;
  -webkit-transition: background-color 300ms ease-out;
  transition: background-color 300ms ease-out;
  padding: 16px 20px!important;
  min-height: 32px;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:hover {
  background-color: rgba(49, 144, 232, 0.05)!important;
}

.emotion-0>button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -16px;
  text-align: inherit!important;
}

.emotion-0>button>div {
  margin: 0 0 0 16px;
}

.emotion-0>button>div:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 {
  cursor: inherit;
  font-weight: 500!important;
  line-height: 1.2!important;
  word-break: break-word!important;
}

.emotion-2 {
  padding: 0.5em 0 0.5em 0.25em!important;
}

.emotion-3 {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  -webkit-align-items: center!important;
  -webkit-box-align: center!important;
  -ms-flex-align: center!important;
  align-items: center!important;
  padding: 0.5em 0 0.5em 0.25em!important;
  color: #3498F4!important;
}

.emotion-3 svg {
  height: 1.5em;
  fill: currentColor;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      data-testid="clickable-notification"
    >
      <div>
        <p
          class="emotion-1"
          title="New Comment: Tables in knowledgebase"
        >
          New Comment: Tables in knowledgebase
        </p>
      </div>
      <div
        class="emotion-2"
      >
        <div
          aria-expanded="false"
        >
          9mo
        </div>
      </div>
    </button>
    <div
      aria-expanded="false"
      class="emotion-3"
      data-testid="icon"
    >
      <svg
        viewBox="0 0 512 512"
      >
        <circle
          cx="256"
          cy="256"
          r="64"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`components ClickableNotification render theming the notification is seen and read renders the notification with the default theme variation 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  background-color: transparent!important;
  border-radius: 8px;
  color: #3A424D!important;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px;
  margin: 4px!important;
  overflow: hidden;
  text-align: left!important;
  text-transform: none;
  -webkit-transition: background-color 300ms ease-out;
  transition: background-color 300ms ease-out;
  padding: 16px 20px!important;
  min-height: 32px;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:hover {
  background-color: rgba(49, 144, 232, 0.15)!important;
}

.emotion-0>button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -16px;
  text-align: inherit!important;
}

.emotion-0>button>div {
  margin: 0 0 0 16px;
}

.emotion-0>button>div:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 {
  cursor: inherit;
  font-weight: 500!important;
  line-height: 1.2!important;
  word-break: break-word!important;
}

.emotion-2 {
  color: inherit!important;
  cursor: inherit;
  margin: 0;
  line-height: 1.2;
  font-size: 0.865em!important;
  margin-top: 1em!important;
  word-break: break-word!important;
}

.emotion-3 {
  padding: 0.5em 0 0.5em 0.25em!important;
}

.emotion-4 {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  -webkit-align-items: center!important;
  -webkit-box-align: center!important;
  -ms-flex-align: center!important;
  align-items: center!important;
  padding: 0.5em 0 0.5em 0.25em!important;
  color: rgba(58, 66, 77, 0.5)!important;
}

.emotion-4 svg {
  height: 1.5em;
  fill: currentColor;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      data-testid="clickable-notification"
    >
      <div>
        <p
          class="emotion-1"
          title="New Comment: Tables in knowledgebase"
        >
          New Comment: Tables in knowledgebase
        </p>
        <article
          class="emotion-2"
        >
          <p>
            This is a good content
          </p>
        </article>
      </div>
      <div
        class="emotion-3"
      >
        <div
          aria-expanded="false"
        >
          9mo
        </div>
      </div>
    </button>
    <div
      aria-expanded="false"
      class="emotion-4"
      data-testid="icon"
    >
      <svg
        viewBox="0 0 512 512"
      >
        <circle
          cx="256"
          cy="64"
          r="64"
        />
        <circle
          cx="256"
          cy="256"
          r="64"
        />
        <circle
          cx="256"
          cy="448"
          r="64"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`components ClickableNotification render theming the notification is unseen renders the notification with the unseen theme variation 1`] = `
.emotion-0 {
  -webkit-backface-visibility: visible;
  backface-visibility: visible;
  -webkit-background-clip: border-box;
  background-clip: border-box;
  background-color: transparent;
  background-origin: padding-box;
  -webkit-background-position: 0 0;
  background-position: 0 0;
  -webkit-background-size: auto;
  background-size: auto;
  border-radius: 0;
  border: 0;
  box-shadow: none;
  box-sizing: content-box;
  clear: none;
  clip: auto;
  color: inherit;
  cursor: auto;
  direction: inherit;
  display: block;
  left: auto;
  font-family: -apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen-Sans,Ubuntu,Cantarell,'Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: normal;
  height: auto;
  -webkit-hyphens: manual;
  -moz-hyphens: manual;
  -ms-hyphens: manual;
  hyphens: manual;
  letter-spacing: normal;
  line-height: 1;
  margin: 0;
  max-height: none;
  max-width: none;
  min-height: 0;
  min-width: 0;
  opacity: 1;
  outline: invert none medium;
  padding: 0;
  perspective: none;
  position: static;
  right: auto;
  top: auto;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  visibility: inherit;
  white-space: normal;
  width: auto;
  word-spacing: normal;
  z-index: auto;
  background-color: rgba(52, 152, 244, 0.05)!important;
  border-radius: 8px;
  color: #3A424D!important;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Oxygen-Sans,Ubuntu,Cantarell,"Helvetica Neue",sans-serif!important;
  font-size: 14px;
  margin: 4px!important;
  overflow: hidden;
  text-align: left!important;
  text-transform: none;
  -webkit-transition: background-color 300ms ease-out;
  transition: background-color 300ms ease-out;
  padding: 16px 20px!important;
  min-height: 32px;
}

.emotion-0 div,
.emotion-0 span,
.emotion-0 p,
.emotion-0 a,
.emotion-0 strong,
.emotion-0 b,
.emotion-0 u,
.emotion-0 i,
.emotion-0 ol,
.emotion-0 ul,
.emotion-0 li,
.emotion-0 form,
.emotion-0 label,
.emotion-0 table,
.emotion-0 tr,
.emotion-0 th,
.emotion-0 td,
.emotion-0 article,
.emotion-0 aside,
.emotion-0 footer,
.emotion-0 header,
.emotion-0 nav,
.emotion-0 section {
  margin: 0;
  padding: 0;
  border: 0;
  font-size: 100%;
  font: inherit;
  vertical-align: baseline;
}

.emotion-0 a {
  cursor: pointer!important;
  color: inherit!important;
}

.emotion-0 button {
  background: transparent;
  border: none;
  font-size: inherit!important;
  color: inherit;
  padding: 0;
  cursor: pointer!important;
}

.emotion-0:hover {
  background-color: rgba(49, 144, 232, 0.1)!important;
}

.emotion-0>button {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 0 0 -16px;
  text-align: inherit!important;
}

.emotion-0>button>div {
  margin: 0 0 0 16px;
}

.emotion-0>button>div:first-of-type {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-1 {
  cursor: inherit;
  font-weight: 500!important;
  line-height: 1.2!important;
  word-break: break-word!important;
}

.emotion-2 {
  color: inherit!important;
  cursor: inherit;
  margin: 0;
  line-height: 1.2;
  font-size: 0.865em!important;
  margin-top: 1em!important;
  word-break: break-word!important;
}

.emotion-3 {
  padding: 0.5em 0 0.5em 0.25em!important;
}

.emotion-4 {
  display: -webkit-box!important;
  display: -webkit-flex!important;
  display: -ms-flexbox!important;
  display: flex!important;
  -webkit-align-items: center!important;
  -webkit-box-align: center!important;
  -ms-flex-align: center!important;
  align-items: center!important;
  padding: 0.5em 0 0.5em 0.25em!important;
  color: #3498F4!important;
}

.emotion-4 svg {
  height: 1.5em;
  fill: currentColor;
}

<div>
  <div
    class="emotion-0"
  >
    <button
      data-testid="clickable-notification"
    >
      <div>
        <p
          class="emotion-1"
          title="New Comment: Tables in knowledgebase"
        >
          New Comment: Tables in knowledgebase
        </p>
        <article
          class="emotion-2"
        >
          <p>
            This is a good content
          </p>
        </article>
      </div>
      <div
        class="emotion-3"
      >
        <div
          aria-expanded="false"
        >
          9mo
        </div>
      </div>
    </button>
    <div
      aria-expanded="false"
      class="emotion-4"
      data-testid="icon"
    >
      <svg
        viewBox="0 0 512 512"
      >
        <circle
          cx="256"
          cy="256"
          r="64"
        />
      </svg>
    </div>
  </div>
</div>
`;
